compiler/ir/backend.jvm/entrypoint/src/org/jetbrains/kotlin/backend/jvm/JvmIrCodegenFactory.kt

```diff
@@ -120,7 +120,7 @@ open class JvmIrCodegenFactory(
             input.languageVersionSettings,
             Psi2IrConfiguration(
                 input.ignoreErrors,
-                partialLinkageEnabled = false,
+                partialLinkageEnabled = input.skipBodies,
```

Comment:
This now generates unbound symbols for code such as:

```
val x by lazy {
    object : Runnable {
            override fun run() {}
        }
}
```

This is covered by the test case `lazyProperty` already in the repo. Unclear to me how to avoid that, so I allowed partial linkage if we skip bodies.

Reply:
It's probably because the type of the property is still generated as the anonymous object. Frontend approximates types of such properties to the anonymous object's supertype, but not for _private_ properties.

I wonder if the same happens with non-lazy properties in kapt + JVM IR? I.e. `private val x = run { object {} }`. If not, I'd prefer to investigate how it works and implement it in the same way for delegated properties.

Reply:
Hmm, that only works for the `by lazy { }` case I guess. If we have `val x by object: { ... getValue ... setValue }` the anonymous type will not be a type parameter.

Reply:
OK, for KAPT, the current attempt approximates anonymous object types to their super type. For types that are not anonymous object types we replace type parameters with star projections to avoid unbound symbols through there. The types are output in erased form anyway.

This fixed a couple of other tests as well.

@udalov can you think of better ways to achieve this?

Reply:
I think this is good enough for now. It seems to work the same in the old backend. I'll merge if remote run is good.